pkg/option/config.go

```diff
@@ -2383,6 +2386,9 @@ type DaemonConfig struct {
 	// compatible with MetalLB's configuration.
 	BGPConfigPath string
 
+	// BGPSecretsNamespace
```

Comment:
I think this comment needs to be fleshed out still.

Reply:
+1

I would expect the secret to reside in the same ns as Cilium (`kube-system`) by default. @dgl why does BGP CP auth require a separate ns from the rest of the Cilium components?

Reply:
A `CiliumBGPPeeringPolicy` is namespace-less, so it seemed strange to me for something that is namespace-less to refer back to  a hardcoded `kube-system`, users shouldn't need to put anything in kube-system in normal operations.

Also this follows the design used by other components that need a secret like ingresses, envoy, etc. Those are that way because those components create secrets, which the BGP CP currently doesn't do but it seemed sensible to match the pattern.

Reply:
Updated the comment.